ann/src/main/java/com/twitter/ann/faiss/NativeUtils.java

```
@@ -90,24 +90,26 @@ public static void unpackLibraryFromJar(String path) throws IOException {
    * @throws FileNotFoundException    If the file could not be found inside the
    *                                  JAR.
    */
-  public static void loadLibraryFromJar(String path) throws IOException {
-    File temp = unpackLibraryFromJarInternal(path);
-
-    try (InputStream is = NativeUtils.class.getResourceAsStream(path)) {
-      Files.copy(is, temp.toPath(), StandardCopyOption.REPLACE_EXISTING);
-    } catch (IOException e) {
-      temp.delete();
-      throw e;
-    } catch (NullPointerException e) {
-      temp.delete();
-      throw new FileNotFoundException("File " + path + " was not found inside JAR.");
+      public static void loadLibraryFromJar(String path) throws IOException {
+        File tempFile = unpackLibraryFromJarInternal(path);
+        try {
+            copyResourceToFile(path, tempFile);
+            System.load(tempFile.getAbsolutePath());
+        } finally {
+            tempFile.deleteOnExit();
+        }
     }
 
-    try {
-      System.load(temp.getAbsolutePath());
-    } finally {
-      temp.deleteOnExit();
-    }
+  private static void copyResourceToFile(String resourcePath, File file) throws IOException {
+      try ( InputStream inputStream = NativeUtils.class.getResourceAsStream(resourcePath)) {
+          Files.copy(inputStream, file.toPath(), StandardCopyOption.REPLACE_EXISTING);
+      } catch (IOException e) {
+          file.delete();
+          throw e;
+      } catch (NullPointerException e) {
+          file.delete();
+          throw new FileNotFoundException("File " + resourcePath + " was not found inside JAR.");
+      }
```

How about creating a catch-all block which catches all other exceptions? 

```
catch (Exception e) {
    file.delete();
    throw e;
}
```